model/task_lifecycle.go

```diff
@@ -472,6 +472,7 @@ func MarkEnd(t *task.Task, caller string, finishTime time.Time, detail *apimodel
 	}
 	if hasFailedTests {
 		detailsCopy.Status = evergreen.TaskFailed
+		detailsCopy.HasFailedTest = true
```

Comment:
I'm not sure this is really the fix; this only affects the color that we display on the new UI really, but the API will be inconsistent. We want to instead fix the underlying problem of having the test failure come from  Maybe the problem is that we always set the failure type. If we instead were to only set it if status=failed, then when we get to this point we'll just have a regular test failure.
https://github.com/evergreen-ci/evergreen/blob/70a14e5170f8a816ee2452dd40f516389cc50ebe/agent/agent.go#L663

Reply:
I think I found a way to fix it.